### PR TITLE
fix: Auth DS re-authorise issue fixed

### LINKED ISSUE #19933: [Bug]: Google OAuth 2.0 cannot re-authorize
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In Edit Datasource, I try to click Save and Re-Authorize, but the button is disabled.

### Steps To Reproduce

I added a datasource for Google Classroom, using Google OAuth 2.0 granting an authorization token. When I save and re-authorize, it works as expected. However, if I wait a few days, it appears that the authorization expires and I need to re-authorize. A few weeks ago I could go into Edit Datasource, scroll to the bottom, and click Save and Re-Authorize. When I tried today, the button is disabled unless a change is made to the configuration.

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR fixes:

Issue with disabled save and authorise button on authenticated API datasource page. When authenticated API is created for authentication type OAuth2, we can see save and authorise button at the bottom of datasource configuration page. When editing existing Auth API datasource, if we do not make any changes in datasource config, button is disabled and user is unable to simply re-authorise their datasource.

> Add a TL;DR when description is extra long (helps content team)

Fixes #19933 
> if no issue exists, please create an issue and ask the maintainers about this first

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      );
+      dataSources.SaveDatasource();
+    });
+
+    dataSources.EditDatasource();
+    agHelper.AssertElementEnabledDisabled(
+      dataSources._saveAndAuthorizeDS,
+      0,
+      false,
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:32]: This is only for `OAuth 2.0` ? or any datasource - when trying to edit - without making any edit on the connection page - will now enable the Save/Test buttons?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+
+let dsName: string;
+const agHelper = ObjectsRegistry.AggregateHelper,
+  dataSources = ObjectsRegistry.DataSources;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:5]: Try to import from ObjectsCore:
Eg:
`import * as _ from "../../../../support/Objects/ObjectsCore";` & then use it like:
`_.dataSources.NavigateToDSCreateNew();`
So no more separate objects declaration is needed in spec files.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts (hunk 1 above)
Comment (sneha122) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:32]: Only for OAuth2.0. Also it does not enable save or test functions but rather save and authorise button that appears when you select authentication type as OAuth2 in Authenticated API creation.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts (hunk 2 above)
Comment (sneha122) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:5]: @Aishwarya-U-R In all other cypress tests written in ts, we have used objectRegistery, is the objectcore something new we have introduced?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts (hunk 1 above)
Comment (sneha122) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:32]: Cc: @Aishwarya-U-R

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts (hunk 2 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/BugTests/Bug19933_spec.ts:5]: Not exactly new, been there for while with less adoption. We are trying to enforce to use it going forward. You can refer to ClientSide/SettingsPane folder for working example.

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good.

**Review by Irongade — APPROVED** [review]

**Pull request merged**